src/components/SearchList.jsx

```diff
@@ -0,0 +1,67 @@
+import { useMemo, useState, useEffect } from 'react';
+import { useHistory, useParams } from 'react-router';
+
+import Link from './Link';
+import data from '../common/axios-config';
+import { BASE_URL, ERRORS } from '../common/constants';
+
+const SearchList = () => {
+  const [searchResults, setSearchResults] = useState([]);
+  const { location } = useParams();
+  const history = useHistory();
+
+  useEffect(() => {
+    const cancelTokenSource = data.CancelToken.source();
+    try {
+      // get all cities that match the search
+      data
+        .get(`${BASE_URL}/location/search/${location}`, {
+          cancelToken: cancelTokenSource.token
+        })
+        .then(({ data: { locations } }) => {
+          setSearchResults(locations);
+        });
+    } catch (error) {
+      alert(ERRORS.DEFAULT.message);
+    }
+
+    return () => cancelTokenSource.cancel();
+  }, [location]);
+
+  const handleSelectCity = e => {
+    // find the exact city chosen from the search results list
+    const selectedCityId = e.target.getAttribute('data-target');
+    const selectedCityName = e.target.innerText;
+
+    // update the url with the searched city name & its id
+    history.push(`/home?city=${selectedCityName}&id=${selectedCityId}`);
+  };
+
+  const matchedResults = useMemo(
```

Comment:
for me, better all business logic move to service. @AntonVlasik or @AV29 what do you think?